dev_guide/pod_preset.adoc

```diff
@@ -0,0 +1,566 @@
+<3> Add a label selector that matches the label in the pod specification.
+<4> Create an environment variable to pass to the container.
+<5> Add a ConfigMap to the pod specification.
+<6> Add a secrets object to the pod specification.
+<7> Specify where external storage volumes should be mounted within the container.
+<8> Define storage volumes that are available to the container(s).
+
+//tag::pod-preset-sample[]
+.Sample pod specification
+----
+apiVersion: v1
+kind: Pod
+metadata:
+  name: website
+  labels:
+    app: website
+    role: frontend <1>
+  annotations:
+    podpreset.admission.kubernetes.io/exclude: "true" <2>
+spec:
+  containers:
+    - name: website
+      image: ecorp/website
+      ports:
+        - containerPort: 80
+----
+
+<1> Add a label to the pod specification that matches the label selector in the pod preset.
+<2> Annotation to prevent the pod preset from modifying a pod spec.
+
+.Sample pod specification after a pod preset
+----
+apiVersion: v1
+kind: Pod
+metadata:
+  name: website
+  labels:
+    app: website
+    role: frontend
+  annotations:
+    podpreset.admission.kubernetes.io/allow-database: "resource version" <1>
+spec:
+  containers:
+    - name: website
+      image: ecorp/website
+      volumeMounts: <2>
+        - mountPath: /cache
+          name: cache-volume
+      ports:
+        - containerPort: 80
+      env: <3>
+        - name: DB_PORT
+          value: "6379"
+      envFrom: <4>
+        - configMapRef:
+          name: etcd-env-config
+        - secretKeyRef:
+          name: test-secret
+  volumes: <5>
+    - name: cache-volume
+      emptyDir: {}
+----
+
+<1> The annotation added to show a pod preset was injected, if the pod specification was not configured to prevent the modification.
+<2> The volume mount is added to the pod.
+<3> The environment variable is added to the pod.
+<4> The ConfigMap and secrets object added by the pod preset.
+<5> The volume mount is added to the pod.
+
+In order to use pod presets in your cluster you must:
+
+* Ask an administrator to enable the pod preset admission controller plug-in through the *_/etc/origin/master/master-config.yaml_*;
+* Enable the api type `settings.k8s.io/v1alpha1/podpreset` through the pod preset and add injectable information to the pod preset.
+//end::pod-preset-sample[]
+
+[[dev-guide-pod-presets-create]]
+== Creating Pod Presets
+
+The following example demonstrates how to create and use pod presets.
+
+[[dev-guide-pod-presets-adm]]
+===== Add the Admission Controller
+
+An administrator can check the *_/etc/origin/master/master-config.yaml_* file to make sure the pod preset admission controller plug-in is present. If
+the admission controller is not present add the plug-in using the following:
+
+----
+admissionConfig:
+  pluginConfig:
+    PodPreset:
+      configuration:
+        kind: DefaultAdmissionConfig
+        apiVersion: v1
+        disable: false# systemctl restart atomic-openshift-master.service
```

Comment:
disable: false# systemctl restart atomic-openshift-master.service
       disable: false

the follow part is not needed here, if it is done in next line
# systemctl restart atomic-openshift-master.service